clippy_lints/src/matches/redundant_guards.rs

```diff
@@ -112,7 +112,7 @@ fn check_method_calls<'tcx>(
     let ty = cx.typeck_results().expr_ty(recv).peel_refs();
     let slice_like = ty.is_slice() || ty.is_array();
 
-    let sugg = if method == sym!(is_empty) {
+    let sugg = if method.as_str() == "is_empty" {
```

Comment:
There's a lot of repeated .as_str() calls on the same parameter here, it might make sense to call it once at the callsite and pass the method as a &str, but this is preexisting code, so ... 🤷
You can leave it if you don't want to change it

Reply:
Passed in the method as &str, nice catch!